### DISCUSSION TITLE
[Discussion] Integrating Papermerge in a third party app

### DISCUSSION THREAD

**Discussion by maheshj01** [ciur/papermerge#565] [General]
Hello @ciur,
Thank you so much for building this great tool. I am looking at papermerge as a potential solution to integrate into our utility mobile app. Our app has users of different roles Building owner, tenant, and contractor for each building. We would like to integrate paper into allowing our users to upload their files and retrieve them for preview.
I looked into the documentation but I couldn't understand whether our requirement could be met, Here are some general questions I have about paper merge, It would be great if you could help clarify the following questions.

- Can we change the permissions of the uploaded file/ control the access of the file, e.g if a contractor uploads it, the tenant should not be able to view the file but the owner should. Is this possible?
- Do we need to deploy an instance of papermerge on our backend in order to consume the rest apis?
- Where are the uploaded files stored, how long do the files stay and what is the storage limit and file upload limit?

Best,
Mahesh

**Comment by ciur** [comment]
I will assume that you are interested in P3 (version 3.0 and later, which will be out December 2023).

> Can we change the permissions of the uploaded file/ control the access of the file, e.g if a contractor uploads it, the tenant should not be able to view the file but the owner should. Is this possible?

Short answer is: yes you can.
Long answer: from P3 application does not manage any authorization/authentication - it just receives a token (which contains currently logged in user), assumes session is valid, and returns you whatever you are asking for.
The AA (authorization and authentication) is managed by other, external application.
Out of the box, P3 comes with its own [AA](https://github.com/papermerge/auth-server/) which currently does not support any permission management at all, but supports multiple authentication schemes.
It is up to you and your particular need to supply the AA part.


> Do we need to deploy an instance of papermerge on our backend in order to consume the rest apis?

Not necessary, papermerge can be used as external service.

> Where are the uploaded files stored, how long do the files stay and what is the storage limit and file upload limit?

Currently the files will be uploaded to the same storage filesystem where application runs, but in general you can customize that as well to support e.g. S3 storage.

**Comment by maheshj01** [comment]
Thanks for the info @ciur

**Comment by maheshj01** [comment]
@ciur I am trying to setup a local server by referring the docs [complete-stack-in-2-minutes](https://docs.papermerge.io/Setup/docker-compose.html#complete-stack-in-2-minutes) 

I was able to run the papermerge client and uploaded few image files and a pdfs, but I am not able to preview any of the images, I tried multiple images of different formats png, jpg,jpeg

<img width="1167" alt="image" src="https://github.com/ciur/papermerge/assets/31410839/5e2153e4-92f9-48e0-bfd4-c51acd4ee3c7">


<img width="1440" alt="image" src="https://github.com/ciur/papermerge/assets/31410839/77c89e24-da57-4661-a610-267671349163">

Any idea what could potentially be wrong here?

Also for consuming the rest APIs do I need to run a different server?

**Comment by ciur** [comment]
yes, the thing is that in version 2.1.x (I guess this is the version you are testings) there is no support for png, jpg and tiff.
Version 2.1.x has only support for pdf files.
Documentation is misleading because prior versions (2.0.x) has support for images.

Support for png, jpg, tiff will be reintroduced in version 3.0.
